File: 1432_Max_Difference_You_Can_Get_From_Changing_an_Integer.java
class Solution {
    public int maxDiff(int num) {
        String stNum = String.valueOf(num);

        // Maximize
        char[] maxValList = stNum.toCharArray();
        char replForMax = ' ';
        for (char charM : maxValList) {
            if (charM != '9') {
                replForMax = charM;
                break;
            }
        }
        
        int maxRes;
        if (replForMax == ' ') {
            maxRes = num;
        } else {
            StringBuilder sbMax = new StringBuilder();
            for (char chM : maxValList) {
                if (chM == replForMax) {
                    sbMax.append('9');
                } else {
                    sbMax.append(chM);
                }
            }
            maxRes = Integer.parseInt(sbMax.toString());
        }

        // Minimize
        char[] minValList = stNum.toCharArray();
        char replForMin = ' ';
        
        int minRes;
        if (minValList[0] != '1') {
            replForMin = minValList[0];
            StringBuilder sbMin = new StringBuilder();
            for (char chMin : minValList) {
                if (chMin == replForMin) {
                    sbMin.append('1');
                } else {
                    sbMin.append(chMin);
                }
            }
            minRes = Integer.parseInt(sbMin.toString());
        } else {
            for (int iMin = 1; iMin < minValList.length; iMin++) {
                if (minValList[iMin] != '0' && minValList[iMin] != '1') {
                    replForMin = minValList[iMin];
                    break;
                }
            }
            if (replForMin == ' ') {
                minRes = num;
            } else {
                StringBuilder sbMin = new StringBuilder();
                for (char chMin : minValList) {
                    if (chMin == replForMin) {
                        sbMin.append('0');
                    } else {
                        sbMin.append(chMin);
                    }
                }
                minRes = Integer.parseInt(sbMin.toString());
            }
        }
        
        return maxRes - minRes;
    }
}

File: 1432_Max_Difference_You_Can_Get_From_Changing_an_Integer.py
class Solution(object):
    def maxDiff(self, num):
        st_num = str(num)

        # Maximize
        max_val_list = list(st_num)
        repl_for_max = ''
        for char_m in max_val_list:
            if char_m != '9':
                repl_for_max = char_m
                break
        
        if repl_for_max == '':
            max_res = num
        else:
            max_res = int("".join(['9' if ch_m == repl_for_max else ch_m for ch_m in max_val_list]))

        # Minimize
        min_val_list = list(st_num)
        repl_for_min = ''
        
        if min_val_list[0] != '1':
            repl_for_min = min_val_list[0]
            min_res = int("".join(['1' if ch_min == repl_for_min else ch_min for ch_min in min_val_list]))
        else:
            for i_min in range(1, len(min_val_list)):
                if min_val_list[i_min] != '0' and min_val_list[i_min] != '1':
                    repl_for_min = min_val_list[i_min]
                    break
            if repl_for_min == '':
                min_res = num
            else:
                min_res = int("".join(['0' if ch_min == repl_for_min else ch_min for ch_min in min_val_list]))
        
        return max_res - min_res
